src/main/scala/com/mwz/sonar/scala/sensor/Measures.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala.sensor
+
+import scala.annotation.tailrec
+import scalariform.lexer.{Token, Tokens}
+
+/**
+ * Scala Sensor Metrics.
+ *
+ * @author mwz
+ * @author BalmungSan
+ */
+object Measures {
+  def countClasses(tokens: List[Token]): Int = {
+    tokens.foldLeft(0) {
+      case (acc, token) =>
+        val tokenType = token.tokenType
+        if (tokenType == Tokens.CLASS || tokenType == Tokens.OBJECT) acc + 1
+        else acc
+    }
+  }
+
+  def countMethods(tokens: List[Token]): Int = {
+    tokens.foldLeft(0) {
+      case (acc, token) =>
+        if (token.tokenType == Tokens.DEF) acc + 1
+        else acc
+    }
+  }
+
+  /* applied on raw source code */
+
+  /* applied on lines of code */
+
+  /* applied on tokenized code */
+
+  @tailrec
+  def countCommentLines(tokens: List[Token], i: Int = 0): Int = {
+    tokens match {
+      case Nil => i
+      case token :: tail if token.tokenType.isComment =>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Pattern match align [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.sonar-scala.com/coding_rules#rule_key=Scalastyle%3Ascalastyle_PatternMatchAlignChecker)

Reply:
@mwz If we plan to have this rule, we must configure scalafmt to do it automatically.
Its funny because my editor does, but when I compiled the project, scalafmt removed the alignment.

Reply:
I've disabled this rule, because it doesn't work well if your cases are not inline, so in this case even if you moved `i` to the next line you would still need to align the arrows, which would be pretty ugly.
```scala
case Nil                                        => i
case token :: tail if token.tokenType.isComment =>
  token.tokenType match {
    ...
``` 
or
```scala
case Nil                                        =>
  i
case token :: tail if token.tokenType.isComment =>
  token.tokenType match {
    ...
```

Reply:
Yeah I agree with you that in many cases the alignment rule result in pretty ugly code.